modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/utils.ts

```diff
@@ -318,6 +341,46 @@ function getUncategorizedPredefinedPosition(
 	return {x, y};
 }
 
+export function getUnsupportedRelationshipErrorMessage(
+	nodes: Node<ObjectDefinitionNodeData>[],
+	sourceNode: Node<ObjectDefinitionNodeData>,
+	targetNode: Node<ObjectDefinitionNodeData>
+) {
+	if (sourceNode.data?.system && targetNode.data?.system) {
+		return Liferay.Language.get('system-objects-cannot-be-related');
+	}
+	else if (sourceNode.data?.linkedObjectDefinition) {
+		return Liferay.Language.get(
+			'adding-relationships-directly-to-linked-objects-is-currently-not-allowed'
+		);
+	}
+	else if (
+		sourceNode.data?.modifiable === false &&
+		targetNode.data?.modifiable === false
+	) {
+		return Liferay.Language.get('unmodifiable-objects-cannot-be-related');
+	}
+	else if (
+		sourceNode.data?.storageType === 'salesforce' ||
+		targetNode.data?.storageType === 'salesforce'
+	) {
+		return Liferay.Language.get(
+			'salesforce-objects-do-not-support-relationships'
+		);
+	}
+	else if (
+		checkPostalAddressUnsupportedRelationship(nodes, sourceNode, targetNode)
+	) {
+		return (
+			Liferay.Language.get('not-possible-to-create-this-relationship') +
+			'&nbsp' +
+			'<a class="alert-link" href="https://learn.liferay.com/web/guest/w/dxp/building-applications/objects/creating-and-managing-objects/using-system-objects-with-custom-objects/accessing-accounts-data-from-custom-objects" target="_blank">' +
+			Liferay.Language.get('learn-more') +
+			'</a>'
```

Comment:
can you please open a thread to frontend infra to ask if there is a better way to do this?

Reply:
Hey Rebecca! I opened Thread in FE Infra and they suggested using a component called Learn Message. Researching its uses I found a [perfect example to follow](https://github.com/brianchandotcom/liferay-portal/pull/138607/files). But I still have some doubts, I'm studying the code to understand some gaps.